cfme/tests/cloud/test_keypairs.py

```diff
@@ -93,3 +92,42 @@ def test_keypair_create_cancel(openstack_provider):
 
     with pytest.raises(KeyPairNotFound):
         navigate_to(keypair, 'Details')
+
+
+def test_keypair_add_and_remove_tag(openstack_provider):
+    """ This will test whether it will add and remove tag for newly created Keypair or not
+    and then deletes it.
+
+    Steps:
+        * Provide Keypair name.
+        * Select Cloud Provider.
+        * Add tag to Keypair.
+        * Remove tag from Keypair
+        * Also delete it.
+    """
+    tag = ('Department', 'Accounting')
+    keypair = KeyPair(name=fauxfactory.gen_alphanumeric(), provider=openstack_provider)
+    try:
+        keypair.create()
+    except TimedOutError:
+        if BZ(1444520, forced_streams=['5.6', '5.7', 'upstream']).blocks:
+            pytest.skip('Timed out creating keypair, BZ1444520')
+        else:
+            pytest.fail('Timed out creating keypair')
+    keypair.exists()
+
+    keypair.add_tag(tag)
+    tagged_value = mixins.get_tags(tag="My Company Tags")
+    assert tuple(tagged_value[0].split(": ", 1)) == tag, "Add tag failed."
```

Comment:
Awesome, its a little messy but I think its still plenty maintainable. Consider adding a comment in-line describing the conversion here (something like "# get_tags returns string formatted 'category: tag', extract the tag portion"